alphastats/gui/utils/preprocessing_helper.py

```diff
@@ -1,144 +1,255 @@
+            "height": 60,
+            "text-valign": "center",
+            "text-halign": "center",
+        },
+    },
+    {
+        "selector": "node.active",
+        "style": {
+            "background-color": "lightgreen",
+        },
+    },
+    {
+        "selector": "node.inactive",
+        "style": {
+            "background-color": "lightgrey",
+        },
+    },
+    {
+        "selector": "node.selected",  # TODO: This currently does not work, figure out why
+        "style": {"background-color": "red"},
+    },
+    {
+        "selector": "edge",
+        "style": {
+            "width": 3,
+            "curve-style": "bezier",
+            "target-arrow-shape": "triangle",
+        },
+    },
+]
+
+
+# TODO: Make help texts meaningful
+# TODO: Show help texts on the widgets
+class PREPROCESSING_STEPS:
+    REMOVE_CONTAMINATONS = {
+        "key": "remove_contaminations",
+        "repr": "Remove contaminations",
+        "help": "Remove contaminations annotated in the contaminations library and filter columns included in the dataset.",
+    }
+    REMOVE_SAMPLES = {
+        "key": "remove_samples",
+        "repr": "Remove samples",
+        "help": "Remove samples from analysis, e.g. useful when failed or blank runs are included.",
+    }
+    SUBSET = {
+        "key": "subset",
+        "repr": "Subset data",
+        "help": "Subset data so it matches with metadata. Can for example be useful if several dimensions of an experiment were analysed together.",
+    }
+    DATA_COMPLETENESS = {
+        "key": "data_completeness",
+        "repr": "Filter data completeness",
+        "help": "Filter data based on completeness across samples. E.g. if a protein has to be detected in at least 70% of the samples.",
+    }
+    LOG2_TRANSFORM = {
+        "key": "log2_transform",
+        "repr": "Log2 transform",
+        "help": "Log2-transform dataset.",
+    }
+    NORMALIZATION = {
+        "key": "normalization",
+        "repr": "Normalization",
+        "help": 'Normalize data using one of the available methods ("zscore", "quantile", "vst", "linear").',
+    }
+    IMPUTATION = {
+        "key": "imputation",
+        "repr": "Imputation",
+        "help": 'Impute missing values using one of the available methods ("mean", "median", "knn", "randomforest").',
+    }
+    BATCH = {
+        "key": "batch",
+        "repr": "Batch correction",
+        "help": "Batch correction.",
+    }
+
+
+PREDEFINED_ORDER = [
+    PREPROCESSING_STEPS.REMOVE_CONTAMINATONS["key"],
+    PREPROCESSING_STEPS.REMOVE_SAMPLES["key"],
+    PREPROCESSING_STEPS.SUBSET["key"],
+    PREPROCESSING_STEPS.DATA_COMPLETENESS["key"],
+    PREPROCESSING_STEPS.LOG2_TRANSFORM["key"],
+    PREPROCESSING_STEPS.NORMALIZATION["key"],
+    PREPROCESSING_STEPS.IMPUTATION["key"],
+    PREPROCESSING_STEPS.BATCH["key"],
+]
+
+
+def draw_workflow(workflow: list[str], order: list[str] = PREDEFINED_ORDER):
+    def find_step_by_key_value(key, target_value):
```

Comment:
hm, why did we make `PREPROCESSING_STEPS` a class? could as well be a `dict` of `dicts` ..

Reply:
https://github.com/MannLabs/alphapeptstats/pull/308/commits/112a6d98626d23b84a3371c5077a0e449365a97d